component/componenttest/error_waiting_host.go

```diff
@@ -12,38 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package extensiontest
+package componenttest
 
 import (
 	"time"
 
 	"github.com/open-telemetry/opentelemetry-collector/component"
 )
 
-// MockHost mocks an component.Host for test purposes.
-type MockHost struct {
+// ErrorWaitingHost mocks an component.Host for test purposes.
+type ErrorWaitingHost struct {
 	errorChan chan error
 }
 
-// NewMockHost returns a new instance of MockHost with proper defaults for most
+var _ component.Host = (*ErrorWaitingHost)(nil)
+
+// NewErrorWaitingHost returns a new instance of ErrorWaitingHost with proper defaults for most
 // tests.
-func NewMockHost() *MockHost {
-	return &MockHost{
+func NewErrorWaitingHost() *ErrorWaitingHost {
+	return &ErrorWaitingHost{
 		errorChan: make(chan error, 1),
 	}
 }
 
 // ReportFatalError is used to report to the host that the extension encountered
 // a fatal error (i.e.: an error that the instance can't recover from) after
 // its start function has already returned.
-func (mh *MockHost) ReportFatalError(err error) {
+func (mh *ErrorWaitingHost) ReportFatalError(err error) {
```

Comment:
nit: rename the receiver pointer according to the renamed type (same for the other renamed type). For the extra nit rename the locals in the tests as appropriate.

Reply:
done.